app/assets/javascripts/app/views/notification_dropdown_view.js

```diff
@@ -131,6 +134,22 @@ app.views.NotificationDropdown = app.views.Base.extend({
       this.dropdownNotifications.perfectScrollbar("destroy");
       this.perfectScrollbarInitialized = false;
     }
+  },
+
+  updateHeaderCounts: function() {
+    $.getJSON(Routes.notificationsCounts(), function(data) {
+      var headerBadge = $(".notifications-link .badge"),
+        markAllReadLink = $("a#mark_all_read_link");
+      if (data.notifications > 0) {
+        headerBadge.html(parseInt(data.notifications));
+        headerBadge.removeClass("hidden");
+        markAllReadLink.removeClass("enabled");
+      } else {
+        headerBadge.addClass("hidden");
+        headerBadge.html(0);
+        markAllReadLink.removeClass("disabled");
+      }
+    });
```

Comment:
We currently handle these kind of changes in the notification view. I think it is a good idea to have them in the notification dropdown view. Maybe write a helper to set the header count which you can call with the number of unread notifications. Then you can use this function from the notifications view. The notification dropdown view should be always accessible under `app.header.notificationDropdown`.